pkg/ottl/ottlfuncs/func_decode_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+func TestDecode(t *testing.T) {
+	type testCase struct {
+		name          string
+		value         any
+		encoding      string
+		want          any
+		expectedError string
+	}
+	tests := []testCase{
```

Comment:
I know we essentially already have this covered, but could we also do a test for a byte array with an IANA encoding?